docs/configuration-files.md

```diff
@@ -197,6 +199,30 @@ rules:
 *Note that single (`'`) and double (`"`) quotes [behave differently](https://docs.octoprint.org/en/master/configuration/yaml.html#scalars)
 in YAML syntax. Single quotes are typically preferred when using backslashes (`\`) with `pattern-regex`.*
 
+### `metavariable-regex`
+
+The `metavariable-regex` operator searches metavariables for a [Python `re.match`](https://docs.python.org/3/library/re.html#re.match)
+compatible expression. This is useful for filtering results based on a [metavariable's](/docs/pattern-features.md#metavariables)
+value.
+
+**Example**
+
+The `metavariable-regex` operator is a mapping which requires the
+`metavariable` and `regex` keys. It can be combined with other pattern operators:
+
+```yaml
+rules:
+  - id: insecure-methods
+    patterns:
+      - pattern: module.$METHOD(...)
+      - metavariable-regex:
+          metavariable: '$METHOD'
```

Comment:
Why the quotes around $METHOD?
Why the quotes around the regexp below? The main advantage of YAML is not having
to use "" as in JSON around strings.
We do not use quotes around semgrep patterns for examples.